libp2p/muxers/muxer.nim

```diff
@@ -46,22 +46,22 @@ proc newMuxerProvider*(creator: MuxerConstructor, codec: string): MuxerProvider
 
 method init(c: MuxerProvider) =
   proc handler(conn: Connection, proto: string) {.async, gcsafe, closure.} =
-    let
-      muxer = c.newMuxer(conn)
+    try:
+      let
+        muxer = c.newMuxer(conn)
 
-    if not isNil(c.streamHandler):
-      muxer.streamHandler = c.streamHandler
+      if not isNil(c.streamHandler):
+        muxer.streamHandler = c.streamHandler
 
-    var futs = newSeq[Future[void]]()
+      var futs = newSeq[Future[void]]()
+      futs &= muxer.handle()
 
-    futs &= muxer.handle()
+      # finally await both the futures
+      if not isNil(c.muxerHandler):
+        futs &= c.muxerHandler(muxer)
 
-    # finally await both the futures
-    if not isNil(c.muxerHandler):
-      futs &= c.muxerHandler(muxer)
-
-    # log and re-raise on errors
-    futs = await allFinished(futs)
-    checkFutures(futs)
+      checkFutures(await allFinished(futs))
+    except CatchableError as exc:
```

Comment:
hmm can anything throw here anyway?

Reply:
If anything does, then it will crash the handler, just good practice.